components/datadog/agent/package.go

```diff
@@ -93,6 +94,14 @@ func GetPackagePath(localPath string, flavor tifos.Flavor, agentFlavor string, a
 				}
 			}
 
+			// If we're on Windows, we need to check if the pipeline ID is in the package name, this is a workaround to avoid an issue with the Windows Runners
+			// where other pipelines packages can pollute the artifacts
+			if flavor == tifos.WindowsServer && pipelineID != "" {
```

Comment:
not all artifacts have the pipeline in their name, see https://github.com/DataDog/datadog-agent/pull/23981